sql/api/src/main/scala/org/apache/spark/sql/streaming/MapState.scala

```diff
@@ -24,30 +24,85 @@ import org.apache.spark.annotation.Evolving
  */
 trait MapState[K, V] extends Serializable {
 
-  /** Whether state exists or not. */
+  /**
+   * Function to check whether any user map entry exists for current grouping key or not.
+   *
+   * @return - true if state exists, false otherwise.
+   */
   def exists(): Boolean
 
-  /** Get the state value if it exists */
+  /**
+   * Get the state value for current grouping key and user map key if it exists or
+   * return null otherwise.
```

Comment:
I'd rather update here (and the same in ValueState.get), it's not null but the default value of the type. For AnyVal they need to follow the default value of the language, and for AnyRef it's simply null.

Once users get this information, they would selectively use the pattern of `Option.getOrElse()` and we wouldn't need to explain this.